src/main/java/uk/ac/ic/wlgitbridge/snapshot/base/MissingRepositoryException.java

```diff
@@ -0,0 +1,51 @@
+package uk.ac.ic.wlgitbridge.snapshot.base;
+
+import com.google.gson.JsonElement;
+import uk.ac.ic.wlgitbridge.git.exception.SnapshotAPIException;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MissingRepositoryException extends SnapshotAPIException {
+
+    public static final List<String> GENERIC_REASON = Arrays.asList(
+        "This Overleaf project currently has no git access.",
+        "",
+        "If this problem persists, please contact us."
+    );
+
+    public static final List<String> EXPORTED_TO_V2 = Arrays.asList(
+        "This Overleaf project has been moved to Overleaf v2, and git access is temporarily unsupported.",
+        "",
+        "See https://www.overleaf.com/help/342 for more information."
+    );
+
+    private List<String> descriptionLines;
+
+    public MissingRepositoryException() {
+        descriptionLines = new ArrayList<String>();
+    }
+
+    public MissingRepositoryException(String message) {
+        this.descriptionLines = Arrays.asList(message);
+    }
+
+    public MissingRepositoryException(List<String> descriptionLines) {
+        this.descriptionLines = descriptionLines;
+    }
+
+    @Override
+    public void fromJSON(JsonElement json) {}
+
+    @Override
+    public String getMessage() {
+        return String.join("\n", this.descriptionLines);
```

Comment:
Looking at other classes, this seems to be a shorter form of the message, e.g. https://github.com/winstonli/writelatex-git-bridge/blob/23ad5c05c7043962619eaf3253015ec09e75cbba/src/main/java/uk/ac/ic/wlgitbridge/snapshot/base/ForbiddenException.java , and it gets used here:

https://github.com/winstonli/writelatex-git-bridge/blob/49d90e2f0dd52ab61aa635fcecfad95dc3a71cb8/src/main/java/uk/ac/ic/wlgitbridge/git/handler/hook/WriteLatexPutHook.java#L97-L119

Will this result in the message being sent twice?

Reply:
I don't think so.  I tried cloning a project, migrating it to v2, locally committing a change, and then pushing.  I just got the one message.

Reply:
OK. From the comments in jgit, it sounds like the client may ignore the message anyway:

https://github.com/eclipse/jgit/blob/f58c3fce90daa92633a17b48feeeeae16db10341/org.eclipse.jgit/src/org/eclipse/jgit/transport/BaseReceivePack.java#L1034

So maybe it does not matter. I guess we will find out.

Reply:
OK. Looking at [the comments in the jgit source](https://github.com/eclipse/jgit/blob/f58c3fce90daa92633a17b48feeeeae16db10341/org.eclipse.jgit/src/org/eclipse/jgit/transport/BaseReceivePack.java#L1034), it says that clients are supposed to display this error, but it also says that there's no guarantee, which is I guess why we use `sendMessage` as well (for the display lines). I would feel somewhat better about being consistent with the other errors. Maybe just return "no git access" or something like that?